source/allocator.c

```diff
@@ -25,30 +25,65 @@
 #    pragma warning(disable : 4100)
 #endif
 
+#ifndef PAGE_SIZE
+#    define PAGE_SIZE (4 * 1024)
```

Comment:
Is there any way we can genuinely determine this per-platform as it's probably a kernel setting?

Reply:
On POSIX, it's defined as PAGE_SIZE. On windows, it's 4K unless you twiddle it yourself.

Reply:
Ok, that's helpful

Reply:
Additionally, you could use madvise() on POSIX to switch to HUGEPAGE, which makes 2MB pages. This is usually best left to the system to worry about. We could also encode this as a separate value that we default to PAGE_SIZE, but can be overridden. Thoughts?

Reply:
Brain buffer overrun to be honest.